packages/tests/webcomponentsjs_/event-listener-options.html

```diff
@@ -0,0 +1,371 @@
+            assert.isTrue(
+              functionCalled,
+              'function called when handleEvent is defined on it'
+            );
+            assert.isFalse(
+              handleEventCalled,
+              'handleEvent not called when on listener function'
+            );
+          });
+
+          test('support `once` option', function () {
+            el1.addEventListener('click', callback, {once: true});
+            el1.click();
+            assert.equal(callback.callCount, 1, 'listener called once');
+            el1.click();
+            assert.equal(callback.callCount, 1, 'respects once option');
+          });
+
+          test('same listener with different `once` options', function () {
+            // same listener adds only once
+            el1.addEventListener('click', callback, {once: true});
+            el1.addEventListener('click', callback);
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 1, 'listener called once');
+            // same listener different event
+            el1.addEventListener('foo', callback, {once: true});
+            el1.addEventListener('foo', callback);
+            el1.dispatchEvent(new Event('foo'));
+            assert.equal(callback.callCount, 2, 'listener called twice');
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 2, 'listener called twice');
+            // can re-add listener
+            el1.addEventListener('click', callback);
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 3, 'called after re-listening');
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 4, 'called when once not used');
+            // changing once state doesn't affect listener
+            el1.addEventListener('click', callback, {once: true});
+            el1.dispatchEvent(new Event('click'));
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 6, 'changing `once` does nothing');
+          });
+
+          test('`once` option can remove listener early and be re-used', function () {
+            el1.addEventListener('click', callback, {once: true});
+            el1.removeEventListener('click', callback);
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 0, 'not called if removed early');
+            el1.addEventListener('click', callback, {once: true});
+            el1.removeEventListener('click', callback, {once: true});
+            el2.addEventListener('click', callback, {once: true});
+            el1.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 0, 'not called if removed early');
+            el2.dispatchEvent(new Event('click'));
+            el2.dispatchEvent(new Event('click'));
+            assert.equal(callback.callCount, 1, 'not called if removed early');
```

Comment:
The assert message feels off for this case.

Reply:
Fixed assert messages.